pkgs/top-level/all-packages.nix

```diff
@@ -6109,7 +6109,6 @@ with pkgs;
 
   tbb_2020_3 = callPackage ../development/libraries/tbb/2020_3.nix { };
   tbb_2021_11 = callPackage ../development/libraries/tbb { };
-  tbb_2022_0 = callPackage ../development/libraries/tbb/2022_0.nix { };
```

Comment:
Instead of removing it and relocating the file I would prefer we change it to this:

```
tbb_2022 = callPackage ../development/libraries/tbb/2022.nix { };
```

The linter complaining is okay to ignore in this case and isn't a blocker. For maintenance and clarity reasons its better to follow the existing convention here.

Once that's changed it should be good to merge.

Reply:
I'm indifferent to whether we go to by-name or not, but I do think that we should do the same thing for all 3.

Dropping the minor version number from the attribute / file name is good.

Reply:
Okay, well then I will leave it up to @Saterfield990.

The reason I was hesitant is because there doesn't seem to be much precedent for named versions under by-name from what I saw at a glance.

But anyway, once all three are consistent whichever way, we should be able to merge.

Reply:
> I have created the following:
> #402508
> #417775

Why not consolidate those into the single PR here? It's easier to track for the current discussion and to reference later if there's a problem.